src/LogViewer/Models/LogRecord.cs

```diff
@@ -34,5 +34,10 @@ public class LogRecord : ModelBase
 
         public bool IsSelected { get; set; }
         #endregion
+
+        public override string ToString()
+        {
+            return $"{DateTime:HH:mm:ss:ms} => [{LogEvent.ToString().ToUpper()}] [{TargetTypeName}] [{ThreadId}] {Message}";
```

Comment:
Cache ToString().ToUpper in a dictionary (use LogEvent as key). We do this in Catel as well and it's a massive performance improvement.